objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -390,6 +399,408 @@ public void sendDataFromImage(ImageRetrieverResults imageResults, ExtensionServi
+                   return;
+               }
+           } else {
+               client.sendQueryError(replyAddress, "io.vantiq.extsrc.objectRecognition.invalidQueryRequest", 
+                       "No imageName or imageDate was specified, or they were incorrectly specified. "
+                       + "Cannot select image(s) to be uploaded.", null);
+               return;
+           }
+       } else {
+           client.sendQueryError(replyAddress, "io.vantiq.extsrc.objectRecognition.invalidQueryRequest", 
+                   "No imageDir was specified, or imageDir was incorrectly specified. "
+                   + "Cannot select image(s) to be uploaded.", null);
+           return;
+       }
+       
+       // Processing the query with all the extracted parameters
+       processImageQueryParameters(imageDir, imageName, imageDate, dateRange, imageDateList, imageUtil, replyAddress);
+   }
+   
+   /**
+    * A helper method called by uploadLocalImages, used to process parameters and call methods that upload images
+    * @param imageDir       Directory containing the local images
+    * @param imageName      The imageName query parameter if specified, otherwise null
+    * @param imageDate      The imageDate query parameter if specified, otherwise null
+    * @param dateRange      The dateRange query parameter if specified, otherwise null
+    * @param imageDateList  The imageDate query parameter if specified as list, otherwise null
+    * @param imageUtil      The instantiated ImageUtil class, setup with properties based on the request
+    * @param replyAddress   The replyAddress used to send a response to the query
+    */
+   public void processImageQueryParameters(String imageDir, String imageName, String imageDate, String dateRange, List<String> imageDateList, 
+           ImageUtil imageUtil, String replyAddress) {
+       
+       boolean uploadOne = true; // Used to decide which upload helper method to call
+       String name = null; // Name used if uploadOne is called
+       
+       // Default filter that selects all files
+       FilenameFilter filter = new FilenameFilter() {
+           public boolean accept(File dir, String name) {
+               return true;
+           }
+       };;
+       
+       // ----------------- PROCESSING PARAMETERS AND CREATING FILTERS ------------------
+       
+       if (imageName != null) {
+           // If imageName is "all", then upload all images in the directory
+           if (imageName.equals("all")) {
+               uploadOne = false;
+               
+           // Otherwise, upload the one image matching the imageName.
+           } else {
+               uploadOne = true;
+               name = imageName;
+           }
+       } else if (imageDate != null) {
+           if (imageDate.equals("all")) {
+               uploadOne = false;
+           } else if(dateRange.equals("none")) {
+               uploadOne = true;
+               name = imageDate;
```

Comment:
Here, if our imageDate is always a list, we create a single 
`createFilter(imageDate)` that just constructs the filter itself.

Or just create a new class
```
class DateRangeFilter implements FileNameFilter {
  beforeDate = null
  afterDate = null
   public DateRangeFilter(List<Date> dateRange) {
      // We assume this list has already been checked for size & dates converted
      beforeDate = dateRange[0] // Either (or both) of these could be null...
      afterDate = dateRange[1]
  }

  @Override
  public boolean accept(File dir, String name) {
        fName =format.parse(name)
			return ((beforeDate == null || fName.after(beforeDate)) &&
                             (afterDate == null || fName.before(afterDate))
		}
    }
}
```
Or something like that (not sure if the logic is quite right, but you get the gist.  The notion being that we'll take the imageDate list & convert it to a list of (2) Dates, either of which could be null.  If both are null, then it's the universal filter.  Otherwise, it does what we need.

The code will look very simple as compared to now, but we'll need a nice explanation of how the filter actually works.